src/plugins/data/public/search/search_interceptor/search_interceptor.ts

```diff
@@ -509,6 +512,15 @@ export class SearchInterceptor {
         return response$.pipe(
           takeUntil(aborted$),
           catchError((e) => {
+            // If we aborted (search:timeout advanced setting) and there was a partial response, return it instead of just erroring out
+            if (searchAbortController.isTimeout()) {
+              return from(this.runSearch(request, { ...options, retrieveResults: true })).pipe(
+                tap(() =>
+                  this.handleSearchError(e, request?.params?.body ?? {}, searchOptions, true)
+                ),
+                map(toPartialResponse)
+              );
+            }
```

Comment:
`Search strategy undefined not found` is especially concerning.

Reply:
Good catch! It looks I was using the wrong argument here, it should have been `from(this.runSearch(request, { ...searchOptions` instead of `...options`. I'll update the PR and add a functional test for this parameter, it's crazy to me there wasn't one already.